### PR TITLE
pkg: rm retry.Run, retry.RunWithInfiniteRetry func since been announced Deprecated for a while

### PR DESCRIPTION
**Pull request by ben1009**
### What problem does this PR solve? 
rm retry.Run, retry.RunWithInfiniteRetry func since been announced Deprecated for a while

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by amyangfei** [comment]
Will this PR be cherry-picked to release branches?

**Comment by ben1009** [comment]
> Will this PR be cherry-picked to release branches?

seems to master is enough, since only change a test file

File: pkg/retry/retry.go [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-	"time"
-
-	"github.com/cenkalti/backoff"
-	"github.com/pingcap/errors"
-)
-
-// Run is Deprecated, try to use Do instead.
-// Run retries the specified function on error for at most maxRetries times.
-// It stops retry if the returned error is context.Canceled or context.DeadlineExceeded.
-func Run(initialInterval time.Duration, maxRetries uint64, f func() error) error {
```
Comment (maxshuang) [inline pkg/retry/retry.go:28]: pr title should be "pkg: rm retry.Run, retry.RunWith...." ??

Comment (ben1009) [reply pkg/retry/retry.go:28]: done

**Review by maxshuang — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**